provider/azure/internal/ad/users.go

```diff
@@ -0,0 +1,63 @@
+// NOTE(axw) this file contains a client for a subset of the
```

Comment:
I've added the LICENSE file from go-autorest, and also added a comment at the top of each file stating that the code is based on that.